--- a/src/config/preset.h
+++ b/src/config/preset.h
@@ -4 +4 @@
-	"1", "2", "3", "4", "5", "6", "7", "8", "9",
+	"1", "2", "3", "4", "5", "6", "7", "8", "9", "0", "T",
--- a/src/ext-protocol/ext-workspace.h
+++ b/src/ext-protocol/ext-workspace.h
@@ -104 +104 @@
-								  "5",		  "6", "7", "8", "9"};
+								  "5",		  "6", "7", "8", "9", "0", "T"};
--- a/src/fetch/monitor.h
+++ b/src/fetch/monitor.h
@@ -66,2 +66,2 @@
-	} else if (i > 9) {
-		return 9;
+	} else if (i > LENGTH(tags)) {
+		return LENGTH(tags);
@@ -88,2 +88,2 @@
-	} else if (i > 9) {
-		return 1 << 8;
+	} else if (i > LENGTH(tags)) {
+		return 1 << (LENGTH(tags) - 1);
